core/rendered_connection.js

```diff
@@ -459,6 +459,11 @@ Blockly.RenderedConnection.prototype.onFailedConnect = function(
  */
 Blockly.RenderedConnection.prototype.disconnectInternal_ = function(parentBlock,
     childBlock) {
+  // The input the child block was connected to (if any).
+  var parentInput = parentBlock.getInputWithBlock(childBlock);
+  if (parentInput && !parentInput.isVisible()) {
+    childBlock.rendered = true;
+  }
```

Comment:
I'm worried about whether this may have unintended side effects.

Reply:
I think it looks good! Because if the block is top-level it should definitely be visible right? As long as the [nested disconnection test](https://github.com/google/blockly/blob/9ab6273e4c2fb0cd25004650d0020490294eed50/tests/mocha/block_test.js#L1444) is passing it should be fine.

I'm going to tag this with #4288 though cuz I know that `.rendered` is kinda crazy wrt collapsed blocks.